### DISCUSSION TITLE
Is it possible to "opt out" of Story Layout altogether, so that no styles are added to the body of the preview iframe?

### DISCUSSION THREAD

**Discussion by MichaelAllenWarner** [storybookjs/storybook#28308] [Help]
### Summary

I know that the [Story Layout](https://storybook.js.org/docs/configure/story-layout) option lets you override the default padded `<body>` configuration. However, it does this by putting different combinations of classes on `<body>`. Is there a way to simply "opt out" of `<body>` classes altogether, so that even `.sb-show-main` isn't there?

I'm working on a project right now where I'm using Storybook to build new components for an existing site that somebody else built. To emulate the environment of the existing site, I'm loading all of its CSS into Storybook's `preview-head.html`. But it turns out that the site has some top-margin on `<body>`, and in Storybook it's getting overridden by the `.sb-show-main { margin: 0; }` rule. I'd like to work around this for some of my stories (e.g., page demos). I imagine that I can find a hacky JS way, but it would be nice if I didn't have to. Any ideas?

**Comment by shilman** [comment]
AFAIK there is nothing "built in". However, Storybook makes it possible to override things like the preview head/body, and even the EJS template that generates the contents of `#sb-main` by modifying your `.storybook/main.js` config. So if you're willing to do a little hacking, I'm confident there's a solution.

https://github.com/storybookjs/storybook/blob/next/code/lib/types/src/modules/core-common.ts#L497-L513

Hope that helps!
